### PR TITLE
Add base-mac-address

### PR DESCRIPTION
**Pull request by dplore**
### Change Scope
* Add base-mac-address leaf `/components/component/state/base-mac-address`.
* This change is backwards compatible

### Platform Implementations
* [JunOS - show chassis mac address](https://www.juniper.net/documentation/us/en/software/junos/chassis/topics/ref/command/show-chassis-mac-addresses.html)
* [Arista EOS - show version](https://www.arista.com/en/um-eos/eos-command-line-interface-cli#xx1125082)
* [Nokia SRL - show version](https://infocenter.nokia.com/public/SRLINUX200R6A/index.jsp?topic=%2Fcom.srlinux.interfaceconfig%2Fhtml%2Fcli-interface.html)

### CODE REVIEW

**Review by rolandphung — APPROVED** [review]

**Comment by dplore** [comment]
Reviewed on Sep 27, 2022 OpenConfig Operators meeting.  Will merge on Oct 11, 2022 if there are no objections.

File: release/models/platform/openconfig-platform.yang [hunk 1]
```diff
@@ -542,6 +548,17 @@ module openconfig-platform {
...
         report the same value.";
     }
+
+    leaf base-mac-address {
+      type oc-yang:mac-address;
+      description
+        "This is a MAC address representing the root or primary MAC
+        address for a component.  Only components such as CHASSIS and
+        CONTROLLER_CARD are expected to providea base-mac-address.  The
+        base mac-address for CHASSIS and a PRIMARY CONTROLLER_CARD may
```
Comment (robshakir) [inline release/models/platform/openconfig-platform.yang:558]: Is there a reason that only the primary controller card and not the secondary can have the same `base-mac-address`? Does this imply that when there is a switchover of controller cards there would be a change in base MAC address? (Seems a challenging idea!)

**Review by robshakir — CHANGES_REQUESTED** [review]

File: release/models/platform/openconfig-platform.yang (hunk 1 above)
Comment (dplore) [reply release/models/platform/openconfig-platform.yang:558]: Good question.  I am collecting some more information on the use case so I can clarify appropriately.

File: release/models/platform/openconfig-platform.yang [hunk 2]
```diff
@@ -542,6 +548,17 @@ module openconfig-platform {
         device, one as primary and the other as secondary, should both
         report the same value.";
     }
+
+    leaf base-mac-address {
+      type oc-yang:mac-address;
+      description
+        "This is a MAC address representing the root or primary MAC
+        address for a component.  Only components such as CHASSIS and
```
Comment (mike-albano) [inline release/models/platform/openconfig-platform.yang:556]: If we remove the word "only" this seems to increase the re-usability of the model. As an example Access Point radios also utilize a "base" mac address from which all BSSIDs utilize over-the-air.

File: release/models/platform/openconfig-platform.yang (hunk 1 above)
Comment (dplore) [reply release/models/platform/openconfig-platform.yang:558]: Updated description with use case.

**Review by robshakir — APPROVED** [review]
Thanks for the fix.

**Pull request merged**